logic/src/commonMain/kotlin/com/wire/kalium/logic/data/wireuser/WireUserRepository.kt

```diff
@@ -12,27 +11,37 @@ import com.wire.kalium.network.api.contact.search.ContactSearchRequest
 import com.wire.kalium.network.api.user.details.ListUserRequest
 import com.wire.kalium.network.api.user.details.UserDetailsApi
 import com.wire.kalium.network.api.user.details.qualifiedIds
+import com.wire.kalium.persistence.dao.UserDAO
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 
-
-interface PublicUserRepository {
+interface WireUserRepository {
+    suspend fun searchKnownUsersByNameOrHandleOrEmail(searchQuery: String): Flow<WireUserSearchResult>
     suspend fun searchPublicContact(
         searchQuery: String,
         domain: String,
         maxResultSize: Int? = null
-    ): Either<CoreFailure, PublicUserSearchResult>
+    ): Either<CoreFailure, WireUserSearchResult>
 }
 
-class PublicUserRepositoryImpl(
+class WireUserRepositoryImpl(
+    private val userDAO: UserDAO,
     private val contactSearchApi: ContactSearchApi,
-    private val userApi: UserDetailsApi,
-    private val publicUserMapper: PublicUserMapper = MapperProvider.publicUserMapper()
-) : PublicUserRepository {
+    private val userDetailsApi: UserDetailsApi,
+    private val wireUserMapper: WireUserMapper = MapperProvider.wireUserMapper()
+) : WireUserRepository {
+
+    override suspend fun searchKnownUsersByNameOrHandleOrEmail(searchQuery: String) =
+        userDAO.getUserByNameOrHandleOrEmail(searchQuery)
+            .map {
+                WireUserSearchResult(it.map { userEntity -> wireUserMapper.fromDaoModelToWireUser(userEntity) })
+            }
 
     override suspend fun searchPublicContact(
```

Comment:
Could this then be changed to `searchWireContact` as `Public` is now renamed to `Wire`?